src/libraries/System.Private.CoreLib/src/System/ValueTuple.cs

```diff
@@ -948,15 +935,14 @@ public int CompareTo(ValueTuple<T1, T2, T3, T4> other)
 
         int IStructuralComparable.CompareTo(object? other, IComparer comparer)
         {
-            if (other == null) return 1;
+            if (other is null) return 1;
 
-            if (!(other is ValueTuple<T1, T2, T3, T4>))
+            if (other is not ValueTuple<T1, T2, T3, T4> objTuple)
             {
                 ThrowHelper.ThrowArgumentException_TupleIncorrectType(this);
+                return 1; // work around CS0170: Use of possibly unassigned field
```

Comment:
You can employ a similar approach to the one previously discussed to avoid the workaround here, too, e.g.
```C#
if (other is not null)
{
    if (other is ValueTuple<T1, T2, T3, T4> objTuple)
    {
        int c = comparer.Compare(Item1, objTuple.Item1);
        ...
        return comparer.Compare(Item4, objTuple.Item4);
    }

    ThrowHelper.ThrowArgumentException_TupleIncorrectType(this);
}

return 1;
```

Reply:
Thanks, I hadn't noticed these methods also needed the change.